mempool/src/shared_mempool/tasks.rs

```diff
@@ -509,8 +511,18 @@ pub(crate) async fn process_config_update<V>(
             .reconfig_update(config_update.clone())
     );
 
-    if let Err(e) = validator.write().restart(config_update) {
+    if let Err(e) = validator.write().restart(config_update.clone()) {
         counters::VM_RECONFIG_UPDATE_FAIL_COUNT.inc();
         error!(LogSchema::event_log(LogEntry::ReconfigUpdate, LogEvent::VMUpdateFail).error(&e));
     }
+
+    let consensus_config: anyhow::Result<OnChainConsensusConfig> = config_update.get();
+    if let Err(error) = &consensus_config {
```

Comment:
this seems a good place to do
```
match consensus_config {
  Ok() => ..
  Err() => ..
}
```

Reply:
The reason the match was not used: If we fail to get the onchain consensus config, we instead use the default value (so we want to still proceed with `.unwrap_or_default()` ) in the Err case.

This is tricky though. Does it make sense to use the default or just ignore? (In this case, if a previous config was good, it would continue to use that value.) Epoch manager uses the default when starting round manager: https://github.com/aptos-labs/aptos-core/blob/main/consensus/src/epoch_manager.rs#L720

Reply:
Yes, this is something unexpected that we should alert on. I guess using the previous value also makes sense -- but it's really all unexpected.